chore(learn): audit javascript algorithms and data structures

naomi-lgbt opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

Audits the challenge formatting and syntax to streamline the i18n process (like we did with Responsive Web Design).

I'm creating a separate commit for each block, to facilitate the review flow (hopefully). Drafting the PR now to get some early eyes on it, will push up commits throughout the weekend to finish.

Commit:
chore(learn): audit basic algorithm scripting

Commit:
chore(learn): audit basic data structures

Commit:
chore(learn): audit basic javascript

Commit:
chore(learn): audit debugging

Commit:
chore(learn): audit es6

Commit:
chore(learn): audit functional programming

Commit:
chore(learn): audit intermidate algorithms

Commit:
chore(learn): audit js projects

Commit:
chore(learn): audit object oriented programming

RandellDawson commented:
@nhcarrigan I did not see the following file in your audit yet, but could you delete the `.` after code block `var sandwich = ["peanut butter", "jelly", "bread"]`. so Crowdin will recognize it as a code block.

`curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/store-multiple-values-in-one-variable-using-javascript-arrays.md`

Commit:
chore(learn): audit regex

Commit:
fix(learn): remove stray .

kelvinsanchez15 commented:
@nhcarrigan In the file: `curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-inequality-operator.md`

Could you change: 
Add the inequality operator `!=` in the `if` statement so that the function will return "Not Equal" when `val` is not equivalent to `99`
With:
Add the inequality operator `!=` in the `if` statement so that the function will return `Not Equal` when `val` is not equivalent to `99`

I can't make the suggestion in that part of the file.

Commit:
fix(learn): string to code

nhcarrigan commented:
Both comments are resolved, and rebase performed to fix conflicts.

kelvinsanchez15 commented:
Nice @nhcarrigan I consider this PR very important for i18n, since it often happens that translated strings are lost from Crowdin's memory when there is a change.

Commit:
fix(learn): missed some

Commit:
fix(learn): clarify strings

Based on Randy's feedback, clarifies string instances where quotes
were removed in favour of back ticks.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/access-an-arrays-contents-using-bracket-notation.md:
```diff
@@ -16,47 +16,47 @@ When we define a simple array as seen below, there are 3 items in it:
 
 # --instructions--
 
 In order to complete this challenge, set the 2nd position (index `1`) of `myArray` to anything you want, besides `"b"`.
```
```suggestion
In order to complete this challenge, set the 2nd position (index `1`) of `myArray` to anything you want, besides the letter `b`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/access-an-arrays-contents-using-bracket-notation.md:
```diff
@@ -16,47 +16,47 @@ When we define a simple array as seen below, there are 3 items in it:
 # --hints--
 
-`myArray[0]` should be equal to `"a"`
+`myArray[0]` should be equal to `a`
```
What about the following hints?  Should they be `to the letter `...`?

nhcarrigan replied:
Letter or string, either one should work.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/add-items-to-an-array-with-push-and-unshift.md:
````diff
@@ -17,12 +17,16 @@ let twentyThree = 'XXIII';
-// now equals ['XIX', 'XX', 'XXI', 'XXII', 'XXIII']Notice that we can also pass variables, which allows us even greater flexibility in dynamically modifying our array's data.
 ```
 
+`romanNumerals` now equals ['XIX', 'XX', 'XXI', 'XXII', 'XXIII']. Notice that we can also pass variables, which allows us even greater flexibility in dynamically modifying our array's data.
````
```suggestion
`romanNumerals` now equals `['XIX', 'XX', 'XXI', 'XXII', 'XXIII']`. Notice that we can also pass variables, which allows us even greater flexibility in dynamically modifying our array's data.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/add-items-using-splice.md:
````diff
@@ -16,11 +16,11 @@ const startIndex = 3;
-// returns [ 10, 11, 12, 13, 14, 15 ]
 ```
 
+The second entry of 12 is removed, and we add 13 and 14 at the same index. The `numbers` array is now [ 10, 11, 12, 13, 14, 15 ].
````
```suggestion
The second entry of `12` is removed, and we add `13` and `14` at the same index. The `numbers` array is now `[ 10, 11, 12, 13, 14, 15 ]`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-all-an-arrays-items-using-for-loops.md:
````diff
@@ -24,18 +24,17 @@ function greaterThanTen(arr) {
 ```
 
-Using a `for` loop, this function iterates through and accesses each element of the array, and subjects it to a simple test that we have created. In this way, we have easily and programmatically determined which data items are greater than `10`, and returned a new array containing those items.
+Using a `for` loop, this function iterates through and accesses each element of the array, and subjects it to a simple test that we have created. In this way, we have easily and programmatically determined which data items are greater than `10`, and returned a new array containing those items. That array is `[12, 14, 80]`
````
```suggestion
Using a `for` loop, this function iterates through and accesses each element of the array, and subjects it to a simple test that we have created. In this way, we have easily and programmatically determined which data items are greater than `10`, and returned a new array containing those items (`[12, 14, 80]`).
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/remove-items-using-splice.md:
````diff
@@ -16,19 +16,20 @@ Ok, so we've learned how to remove elements from the beginning and end of arrays
-// array now equals ['today', 'was', 'great']
 ```
 
+Here we remove 2 elements, beginning with the third element (at index 2). `array` now has the value [`'today', 'was', 'great']`.
````
```suggestion
Here we remove 2 elements, beginning with the third element (at index 2). `array` now has the value `['today', 'was', 'great']`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/use-an-array-to-store-a-collection-of-data.md:
````diff
@@ -13,9 +13,10 @@ The below is an example of the simplest implementation of an array data structur
-// logs 7
 ```
 
+The `console.log` call prints `7`.
````
```suggestion
The `console.log` call displays `7`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/access-array-data-with-indexes.md:
```diff
@@ -19,12 +19,11 @@ Array indexes are written in the same bracket notation that strings use, except
 var array = [50,60,70];
-array[0]; // equals 50
-var data = array[1];  // equals 60
+array[0]; // 50
+var data = array[1];  // 60
```
Remove the comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/access-multi-dimensional-arrays-with-indexes.md:
```diff
@@ -20,13 +20,12 @@ var arr = [
-arr[3]; // equals [[10,11,12], 13, 14]
-arr[3][0]; // equals [10,11,12]
-arr[3][0][1]; // equals 11
+arr[3]; // [[10,11,12], 13, 14]
+arr[3][0]; // [10,11,12]
+arr[3][0][1]; // 11
```
Remove the comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-nested-objects.md:
```diff
@@ -26,7 +26,7 @@ var ourStorage = {
   }
 };
-ourStorage.cabinet["top drawer"].folder2;  // "secrets"
+ourStorage.cabinet["top drawer"].folder2; // "secrets"
 ourStorage.desk.drawer; // "stapler"
```
Remove the comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-variables.md:
```diff
@@ -32,7 +32,7 @@ function propPrefix(str) {
   return s + str;
 }
-var someProp = propPrefix("Name"); // someProp now holds the value 'propName'
+var someProp = propPrefix("Name"); // "propName"
 console.log(someObj[someProp]); // "John"
```
Remove the comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.md:
```diff
@@ -31,38 +31,38 @@ switch (num) {
 # --hints--
 
-`switchOfStuff("a")` should have a value of "apple"
+`switchOfStuff("a")` should have a value of `apple`
```
Do we want to add "the string " before all of these in these challenge hints?

nhcarrigan replied:
That's a good idea, yeah.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-inequality-operator.md:
```diff
@@ -23,35 +23,35 @@ The inequality operator (`!=`) is the opposite of the equality operator. It mean
 # --instructions--
 
-Add the inequality operator `!=` in the `if` statement so that the function will return "Not Equal" when `val` is not equivalent to `99`
+Add the inequality operator `!=` in the `if` statement so that the function will return `Not Equal` when `val` is not equivalent to `99`
```
```suggestion
Add the inequality operator `!=` in the `if` statement so that the function will return the string `Not Equal` when `val` is not equivalent to `99`
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparisons-with-the-logical-and-operator.md:
```diff
@@ -33,7 +33,7 @@ return "No";
 # --instructions--
 
-Replace the two if statements with one statement, using the && operator, which will return `"Yes"` if `val` is less than or equal to `50` and greater than or equal to `25`. Otherwise, will return `"No"`.
+Replace the two if statements with one statement, using the `&&` operator, which will return `Yes` if `val` is less than or equal to `50` and greater than or equal to `25`. Otherwise, will return `No`.
```
```suggestion
Replace the two if statements with one statement, using the `&&` operator, which will return the string `Yes` if `val` is less than or equal to `50` and greater than or equal to `25`. Otherwise, will return the string`No`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/compound-assignment-with-augmented-addition.md:
```diff
@@ -20,7 +20,7 @@ One such operator is the `+=` operator.
 var myVar = 1;
 myVar += 5;
-console.log(myVar); // Returns 6
+console.log(myVar); // 6
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/concatenating-strings-with-plus-operator.md:
````diff
@@ -24,16 +24,16 @@ Example:
 ```js
 var ourStr = "I come first. " + "I come second.";
-// ourStr is "I come first. I come second."
+// "I come first. I come second."
````
Remove this comment from the code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/concatenating-strings-with-the-plus-equals-operator.md:
```diff
@@ -19,16 +19,16 @@ Example:
 var ourStr = "I come first. ";
 ourStr += "I come second.";
-// ourStr is now "I come first. I come second."
+// "I come first. I come second."
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/constructing-strings-with-variables.md:
```diff
@@ -16,12 +16,12 @@ Example:
 var ourName = "freeCodeCamp";
 var ourStr = "Hello, our name is " + ourName + ", how are you?";
-// ourStr is now "Hello, our name is freeCodeCamp, how are you?"
+// "Hello, our name is freeCodeCamp, how are you?"
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/finding-a-remainder-in-javascript.md:
```diff
@@ -20,8 +20,7 @@ In mathematics, a number can be checked to be even or odd by checking the remain
 
-**Note**  
-The <dfn>remainder</dfn> operator is sometimes incorrectly referred to as the "modulus" operator. It is very similar to modulus, but does not work properly with negative numbers.
+**Note:** The <dfn>remainder</dfn> operator is sometimes incorrectly referred to as the "modulus" operator. It is very similar to modulus, but does not work properly with negative numbers.
```
I think the double quotes can be removed around `modulus`.
```suggestion
**Note:** The <dfn>remainder</dfn> operator is sometimes incorrectly referred to as the modulus operator. It is very similar to modulus, but does not work properly with negative numbers.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/local-scope-and-functions.md:
```diff
@@ -18,7 +18,7 @@ function myTest() {
   console.log(loc);
 }
-myTest(); // logs "foo"
+myTest(); // "foo"
```
Remove this comment from the code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulate-arrays-with-pop.md:
```diff
@@ -11,20 +11,20 @@ dashedName: manipulate-arrays-with-pop
 var oneDown = threeArr.pop();
-console.log(oneDown); // Returns 6
-console.log(threeArr); // Returns [1, 4]
+console.log(oneDown); // 6
+console.log(threeArr); // [1, 4]
```
Remove these comments from the code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulate-arrays-with-push.md:
```diff
@@ -18,11 +18,11 @@ Examples:
 var arr1 = [1,2,3];
 arr1.push(4);
-// arr1 is now [1,2,3,4]
+// [1,2,3,4]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulate-arrays-with-push.md:
```diff
@@ -18,11 +18,11 @@ Examples:
 var arr2 = ["Stimpson", "J", "cat"];
 arr2.push(["happy", "joy"]);
-// arr2 now equals ["Stimpson", "J", "cat", ["happy", "joy"]]
+// ["Stimpson", "J", "cat", ["happy", "joy"]]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulate-arrays-with-unshift.md:
````diff
@@ -17,9 +17,9 @@ Example:
 ```js
 var ourArray = ["Stimpson", "J", "cat"];
-ourArray.shift(); // ourArray now equals ["J", "cat"]
+ourArray.shift(); // ourArray: ["J", "cat"]
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulate-arrays-with-unshift.md:
```diff
@@ -17,9 +17,9 @@ Example:
+ourArray.shift(); // ourArray: ["J", "cat"]
 ourArray.unshift("Happy");
-// ourArray now equals ["Happy", "J", "cat"]
+// ourArray: ["Happy", "J", "cat"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/modify-array-data-with-indexes.md:
````diff
@@ -15,19 +15,18 @@ Unlike strings, the entries of arrays are <dfn>mutable</dfn> and can be changed
 ```js
 var ourArray = [50,40,30];
-ourArray[0] = 15; // equals [15,40,30]
+ourArray[0] = 15; // [15,40,30]
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/modify-array-data-with-indexes.md:
```diff
@@ -15,19 +15,18 @@ Unlike strings, the entries of arrays are <dfn>mutable</dfn> and can be changed
 # --hints--
 
-`myArray` should now be [45,64,99].
+`myArray` should now be `[45,64,99]`.
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/multiply-two-numbers-with-javascript.md:
````diff
@@ -16,7 +16,7 @@ JavaScript uses the `*` symbol for multiplication of two numbers.
 
 ```js
-myVar = 13 * 13; // assigned 169
+myVar = 13 * 13; // 169
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/practice-comparing-different-values.md:
````diff
@@ -15,32 +15,28 @@ If the values being compared are not of the same type, the equality operator wil
 ```js
-typeof 3   // returns 'number'
-typeof '3' // returns 'string'
+typeof 3   // 'number'
+typeof '3' // 'string'
````
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.md:
````diff
@@ -9,17 +9,17 @@ dashedName: use-bracket-notation-to-find-the-first-character-in-a-string
 ```js
 var firstName = "Charles";
-var firstLetter = firstName[0]; // firstLetter is "C"
+var firstLetter = firstName[0]; // "C"
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.md:
````diff
@@ -17,7 +17,7 @@ Example:
 ```js
 var firstName = "Charles";
-var lastLetter = firstName[firstName.length - 1]; // lastLetter is "s"
+var lastLetter = firstName[firstName.length - 1]; // "s"
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-last-character-in-a-string.md:
```diff
@@ -28,7 +28,7 @@ Use <dfn>bracket notation</dfn> to find the last character in the `lastName` var
 # --hints--
 
-`lastLetterOfLastName` should be "e".
+`lastLetterOfLastName` should be `e`.
```
```suggestion
`lastLetterOfLastName` should be the letter `e`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.md:
````diff
@@ -17,7 +17,7 @@ Example:
 ```js
 var firstName = "Ada";
-var secondLetterOfFirstName = firstName[1]; // secondLetterOfFirstName is "d"
+var secondLetterOfFirstName = firstName[1]; // "d"
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.md:
````diff
@@ -17,7 +17,7 @@ Example:
 ```js
 var firstName = "Charles";
-var thirdToLastLetter = firstName[firstName.length - 3]; // thirdToLastLetter is "l"
+var thirdToLastLetter = firstName[firstName.length - 3]; // "l"
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-to-last-character-in-a-string.md:
```diff
@@ -28,7 +28,7 @@ Use <dfn>bracket notation</dfn> to find the second-to-last character in the `las
 # --hints--
 
-`secondToLastLetterOfLastName` should be "c".
+`secondToLastLetterOfLastName` should be `c`.
```
```suggestion
`secondToLastLetterOfLastName` should be the letter `c`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-conditional-logic-with-if-statements.md:
```diff
@@ -26,15 +26,15 @@ function test (myCondition) {
 }
-test(true);  // returns "It was true"
-test(false); // returns "It was false"
+test(true);  // "It was true"
+test(false); // "It was false"
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-the-parseint-function.md:
```diff
@@ -13,7 +13,7 @@ The `parseInt()` function parses a string and returns an integer. Here's an exam
 `var a = parseInt("007");`
 
-The above function converts the string "007" to an integer 7. If the first character in the string can't be converted into a number, then it returns `NaN`.
+The above function converts the string `007` to an integer 7. If the first character in the string can't be converted into a number, then it returns `NaN`.
```
```suggestion
The above function converts the string `007` to the integer `7`. If the first character in the string can't be converted into a number, then it returns `NaN`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/use-typeof-to-check-the-type-of-a-variable.md:
```diff
@@ -13,10 +13,10 @@ You can use `typeof` to check the data structure, or type, of a variable. This i
-console.log(typeof 0); // outputs "number"
-console.log(typeof []); // outputs "object"
-console.log(typeof {}); // outputs "object"
+console.log(typeof ""); // "string"
+console.log(typeof 0); // "number"
+console.log(typeof []); // "object"
+console.log(typeof {}); // "object"
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords.md:
```diff
@@ -19,10 +19,8 @@ var numArray = [];
-// returns [0, 1, 2]
-console.log(i);
-// returns 3
+console.log(numArray); // [0, 1, 2]
+console.log(i); // 3
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords.md:
```diff
@@ -33,13 +31,11 @@ var i;
-// returns [0, 1, 2]
-console.log(i);
-// returns 3
+console.log(numArray); // [0, 1, 2]
+console.log(i); // 3
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords.md:
```diff
@@ -65,17 +60,15 @@ for (let i = 0; i < 3; i++) {
-// returns 2
-console.log(i);
-// returns "i is not defined"
+console.log(printNumTwo()); // 2
+console.log(i); // "i is not defined"
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.md:
```diff
@@ -20,11 +20,10 @@ const person = {
 
-console.log(greeting); // prints
+console.log(greeting);
 // Hello, my name is Zodiac Hasbro!
 // I am 56 years old.
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/declare-a-read-only-variable-with-the-const-keyword.md:
````diff
@@ -14,7 +14,7 @@ The keyword `let` is not the only new way to declare variables. In ES6, you can
 ```js
 const FAV_PET = "Cats";
-FAV_PET = "Dogs"; // returns error
+FAV_PET = "Dogs"; // error
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/explore-differences-between-the-var-and-let-keywords.md:
```diff
@@ -13,23 +13,22 @@ One of the biggest problems with declaring variables with the `var` keyword is t
 var camper = 'David';
-console.log(camper);
-// logs 'David'
+console.log(camper); // 'David'
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/explore-differences-between-the-var-and-let-keywords.md:
````diff
@@ -13,23 +13,22 @@ One of the biggest problems with declaring variables with the `var` keyword is t
 ```js
 let camper = 'James';
-let camper = 'David'; // throws an error
+let camper = 'David'; // error
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/explore-differences-between-the-var-and-let-keywords.md:
````diff
@@ -13,23 +13,22 @@ One of the biggest problems with declaring variables with the `var` keyword is t
 ```js
 "use strict";
-x = 3.14; // throws an error because x is not declared
+x = 3.14; // x is not defined
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const.md:
```diff
@@ -16,9 +16,9 @@ However, it is important to understand that objects (including arrays and functi
-s = [1, 2, 3]; // throws error, trying to assign a const
-s[2] = 45; // works just as it would with an array declared with var or let
-console.log(s); // returns [5, 6, 45]
+s = [1, 2, 3]; // error
+s[2] = 45;
+console.log(s); // [5, 6, 45]
```
Remove commens from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/prevent-object-mutation.md:
```diff
@@ -18,8 +18,8 @@ let obj = {
 Object.freeze(obj);
-obj.review = "bad"; // will be ignored. Mutation not allowed
-obj.newProp = "Test"; // will be ignored. Mutation not allowed
+obj.review = "bad"; // error
+obj.newProp = "Test"; // error
 console.log(obj); 
 // { name: "FreeCodeCamp", review:"Awesome"}
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.md:
````diff
@@ -14,20 +14,20 @@ The ES5 code below uses `apply()` to compute the maximum value in an array:
 ```js
 var arr = [6, 89, 3, 45];
-var maximus = Math.max.apply(null, arr); // returns 89
+var maximus = Math.max.apply(null, arr); // 89
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.md:
````diff
@@ -14,20 +14,20 @@ The ES5 code below uses `apply()` to compute the maximum value in an array:
 ```js
 const arr = [6, 89, 3, 45];
-const maximus = Math.max(...arr); // returns 89
+const maximus = Math.max(...arr); // 89
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.md:
````diff
@@ -14,20 +14,20 @@ The ES5 code below uses `apply()` to compute the maximum value in an array:
 
 ```js
-const spreaded = ...arr; // will throw a syntax error
+const spreaded = ...arr; // syntax error
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/write-arrow-functions-with-parameters.md:
```diff
@@ -11,24 +11,21 @@ dashedName: write-arrow-functions-with-parameters
-// doubles input value and returns it
 const doubler = (item) => item * 2;
-doubler(4); // returns 8
+doubler(4); // 8
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/write-arrow-functions-with-parameters.md:
```diff
@@ -11,24 +11,21 @@ dashedName: write-arrow-functions-with-parameters
-// multiplies the first input value by the second and returns it
 const multiplier = (item, multi) => item * multi;
-multiplier(4, 2); // returns 8
+multiplier(4, 2); // 8
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/combine-an-array-into-a-string-using-the-join-method.md:
```diff
@@ -15,12 +15,12 @@ Here's an example:
 var arr = ["Hello", "World"];
 var str = arr.join(" ");
-// Sets str to "Hello World"
+// str == "Hello World"
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/combine-two-arrays-using-the-concat-method.md:
````diff
@@ -12,7 +12,7 @@ dashedName: combine-two-arrays-using-the-concat-method
 ```js
 [1, 2, 3].concat([4, 5, 6]);
-// Returns a new array [1, 2, 3, 4, 5, 6]
+// [1, 2, 3, 4, 5, 6]
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/introduction-to-currying-and-partial-application.md:
```diff
@@ -15,40 +15,36 @@ In other words, it restructures a function so it takes one argument, then return
 const curried = x => y => x + y
 
-curried(1)(2) // Returns 3
+curried(1)(2) // 3
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/introduction-to-currying-and-partial-application.md:
```diff
@@ -15,40 +15,36 @@ In other words, it restructures a function so it takes one argument, then return
-// Call a curried function in parts:
 var funcForY = curried(1);
-console.log(funcForY(2)); // Prints 3
+console.log(funcForY(2)); // 3
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/introduction-to-currying-and-partial-application.md:
```diff
@@ -15,40 +15,36 @@ In other words, it restructures a function so it takes one argument, then return
 }
 var partialFn = impartial.bind(this, 1, 2);
-partialFn(10); // Returns 13
+partialFn(10); // 13
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/return-part-of-an-array-using-the-slice-method.md:
```diff
@@ -15,7 +15,7 @@ Here's an example:
 var arr = ["Cat", "Dog", "Tiger", "Zebra"];
 var newArray = arr.slice(1, 3);
-// Sets newArray to ["Dog", "Tiger"]
+// ["Dog", "Tiger"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/sort-an-array-alphabetically-using-the-sort-method.md:
```diff
@@ -19,15 +19,15 @@ function ascendingOrder(arr) {
 }
 ascendingOrder([1, 5, 2, 3, 4]);
-// Returns [1, 2, 3, 4, 5]
+// [1, 2, 3, 4, 5]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/sort-an-array-alphabetically-using-the-sort-method.md:
```diff
@@ -19,15 +19,15 @@ function ascendingOrder(arr) {
 }
 reverseAlpha(['l', 'h', 'z', 'b', 's']);
-// Returns ['z', 's', 'l', 'h', 'b']
+// ['z', 's', 'l', 'h', 'b']
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/split-a-string-into-an-array-using-the-split-method.md:
```diff
@@ -15,11 +15,11 @@ Here are two examples that split one string by spaces, then another by digits us
 var str = "Hello World";
 var bySpace = str.split(" ");
-// Sets bySpace to ["Hello", "World"]
+// ["Hello", "World"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/split-a-string-into-an-array-using-the-split-method.md:
```diff
@@ -15,11 +15,11 @@ Here are two examples that split one string by spaces, then another by digits us
 var otherString = "How9are7you2today";
 var byDigits = otherString.split(/\d/);
-// Sets byDigits to ["How", "are", "you", "today"]
+// ["How", "are", "you", "today"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-every-method-to-check-that-every-element-in-an-array-meets-a-criteria.md:
```diff
@@ -17,7 +17,7 @@ var numbers = [1, 5, 8, 0, 10, 11];
   return currentValue < 10;
 });
-// Returns false
+// false
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/add-methods-after-inheritance.md:
```diff
@@ -8,7 +8,7 @@ dashedName: add-methods-after-inheritance
 # --description--
 
-A constructor function that inherits its `prototype` object from a supertype constructor function can still have its own methods in addition to inherited methods.
+A `constructor` function that inherits its `prototype` object from a supertype `constructor` function can still have its own methods in addition to inherited methods.
```
Not sure about wrapping "constructor" with backticks.  Would like to see what others think.

nhcarrigan replied:
I should not have wrapped this one - I did wrap the ones that specifically referred to a `constructor()` in the code, but this is not such a case.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/extend-constructors-to-receive-arguments.md:
````diff
@@ -26,7 +26,7 @@ function Bird(name, color) {
+Then pass in the values as arguments to define each unique bird into the `Bird` constructor: `let cardinal = new Bird("Bruce", "red");` This gives a new instance of `Bird` with `name` and `color` properties set to `Bruce` and `red`, respectively. The `numLegs` property is still set to 2. The `cardinal` has these properties:
 
 ```js
 cardinal.name // => Bruce
````
Remove comments from code segment.

ShaunSHamilton replied:
Bringing this up again

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/inherit-behaviors-from-a-supertype.md:
````diff
@@ -32,7 +32,7 @@ let animal = Object.create(Animal.prototype);
 
 ```js
-animal.eat(); // prints "nom nom nom"
+animal.eat(); // "nom nom nom"
 animal instanceof Animal; // => true
````
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/iterate-over-all-properties.md:
```diff
@@ -34,8 +34,8 @@ for (let property in duck) {
 
-console.log(ownProps); // prints ["name"]
-console.log(prototypeProps); // prints ["numLegs"]
+console.log(ownProps); // ["name"]
+console.log(prototypeProps); // ["numLegs"]
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/remember-to-set-the-constructor-property-when-changing-the-prototype.md:
```diff
@@ -11,16 +11,16 @@ dashedName: remember-to-set-the-constructor-property-when-changing-the-prototype
-duck.constructor === Bird; // false -- Oops
-duck.constructor === Object; // true, all objects inherit from Object.prototype
-duck instanceof Bird; // true, still works
+duck.constructor === Bird; // false
+duck.constructor === Object; // true
+duck instanceof Bird; // true
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/set-the-childs-prototype-to-an-instance-of-the-parent.md:
````diff
@@ -20,7 +20,7 @@ Remember that the `prototype` is like the "recipe" for creating an object. In a
 ```js
 let duck = new Bird("Donald");
-duck.eat(); // prints "nom nom nom"
+duck.eat(); // "nom nom nom"
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-own-properties.md:
```diff
@@ -31,7 +31,7 @@ for (let property in duck) {
 }
 
-console.log(ownProps); // prints [ "name", "numLegs" ]
+console.log(ownProps); // [ "name", "numLegs" ]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-constructor-property.md:
```diff
@@ -30,8 +30,7 @@ function joinBirdFraternity(candidate) {
 
-**Note**  
-Since the `constructor` property can be overwritten (which will be covered in the next two challenges) it’s generally better to use the `instanceof` method to check the type of an object.
+**Note:** Since the `constructor` property can be overwritten (which will be covered in the next two challenges) it’s generally better to use the `instanceof` method to check the type of an object.
```
Not sure about wrapping "constructor" here.

nhcarrigan replied:
This refers to the actual `.constructor` property on the object, so IMO should be wrapped.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-prototype-chain.md:
```diff
@@ -15,13 +15,13 @@ function Bird(name) {
 }
 
-typeof Bird.prototype; // yields 'object'
+typeof Bird.prototype; // 'object'
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-prototype-chain.md:
````diff
@@ -15,13 +15,13 @@ function Bird(name) {
 
 ```js
-Object.prototype.isPrototypeOf(Bird.prototype); // returns true
+Object.prototype.isPrototypeOf(Bird.prototype); // true
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-a-mixin-to-add-common-behavior-between-unrelated-objects.md:
````diff
@@ -40,8 +40,8 @@ flyMixin(plane);
 ```js
-bird.fly(); // prints "Flying, wooosh!"
-plane.fly(); // prints "Flying, wooosh!"
+bird.fly(); // "Flying, wooosh!"
+plane.fly(); // "Flying, wooosh!"
````
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.md:
```diff
@@ -23,15 +23,14 @@ The simplest way to make this public property private is by creating a variable
 }
 let ducky = new Bird();
-ducky.getHatchedEggCount(); // returns 10
+ducky.getHatchedEggCount(); // 10
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-dot-notation-to-access-the-properties-of-an-object.md:
```diff
@@ -15,11 +15,10 @@ let duck = {
 };
-console.log(duck.name);
-// This prints "Aflac" to the console
+console.log(duck.name); // "Aflac"
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-prototype-properties-to-reduce-duplicate-code.md:
````diff
@@ -21,8 +21,8 @@ Bird.prototype.numLegs = 2;
 ```js
-console.log(duck.numLegs);  // prints 2
-console.log(canary.numLegs);  // prints 2
+console.log(duck.numLegs);  // 2
+console.log(canary.numLegs);  // 2
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-all-or-none.md:
```diff
@@ -18,13 +18,13 @@ For example, there are slight differences in American and British English and yo
 let rainbowRegex= /colou?r/;
-rainbowRegex.test(american); // Returns true
-rainbowRegex.test(british); // Returns true
+rainbowRegex.test(american); // true
+rainbowRegex.test(british); // true
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md:
```diff
@@ -17,8 +17,7 @@ Then check whether the desired string groups are in the test string by using the
 let testRegex = /P(engu|umpk)in/;
-testRegex.test(testStr);
-// Returns true
+testRegex.test(testStr); // true
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/extract-matches.md:
````diff
@@ -15,12 +15,10 @@ To use the `.match()` method, apply the method on a string and pass in the regex
 ```js
-"Hello, World!".match(/Hello/);
-// Returns ["Hello"]
+"Hello, World!".match(/Hello/); // ["Hello"]
````
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/extract-matches.md:
```diff
@@ -15,12 +15,10 @@ To use the `.match()` method, apply the method on a string and pass in the regex
 let ourRegex = /expressions/;
-ourStr.match(ourRegex);
-// Returns ["expressions"]
+ourStr.match(ourRegex); // ["expressions"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-more-than-the-first-match.md:
```diff
@@ -13,21 +13,19 @@ So far, you have only been able to extract or search a pattern once.
 let ourRegex = /Repeat/;
-testStr.match(ourRegex);
-// Returns ["Repeat"]
+testStr.match(ourRegex); // ["Repeat"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-more-than-the-first-match.md:
```diff
@@ -13,21 +13,19 @@ So far, you have only been able to extract or search a pattern once.
 let repeatRegex = /Repeat/g;
-testStr.match(repeatRegex);
-// Returns ["Repeat", "Repeat", "Repeat"]
+testStr.match(repeatRegex); // ["Repeat", "Repeat", "Repeat"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-a-literal-string-with-different-possibilities.md:
```diff
@@ -8,57 +8,57 @@ dashedName: match-a-literal-string-with-different-possibilities
 This is powerful to search single strings, but it's limited to only one pattern. You can search for multiple patterns using the `alternation` or `OR` operator: `|`.
 
-This operator matches patterns either before or after it. For example, if you wanted to match `"yes"` or `"no"`, the regex you want is `/yes|no/`.
+This operator matches patterns either before or after it. For example, if you wanted to match `yes` or `no`, the regex you want is `/yes|no/`.
```
```suggestion
This operator matches patterns either before or after it. For example, if you wanted to match the strings `yes` or `no`, the regex you want is `/yes|no/`.
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-all-letters-and-numbers.md:
```diff
@@ -17,10 +17,10 @@ let longHand = /[A-Za-z0-9_]+/;
-shortHand.test(numbers); // Returns true
-longHand.test(varNames); // Returns true
-shortHand.test(varNames); // Returns true
+longHand.test(numbers); // true
+shortHand.test(numbers); // true
+longHand.test(varNames); // true
+shortHand.test(varNames); // true
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-beginning-string-patterns.md:
```diff
@@ -15,20 +15,18 @@ In an earlier challenge, you used the caret character (`^`) inside a character s
 let firstRegex = /^Ricky/;
-firstRegex.test(firstString);
-// Returns true
+firstRegex.test(firstString); // true
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-beginning-string-patterns.md:
```diff
@@ -15,20 +15,18 @@ In an earlier challenge, you used the caret character (`^`) inside a character s
 let notFirst = "You can't find Ricky now.";
-firstRegex.test(notFirst);
-// Returns false
+firstRegex.test(notFirst); // false
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-characters-that-occur-zero-or-more-times.md:
```diff
@@ -17,14 +17,14 @@ let soccerWord = "gooooooooal!";
-soccerWord.match(goRegex); // Returns ["goooooooo"]
-gPhrase.match(goRegex); // Returns ["g"]
-oPhrase.match(goRegex); // Returns null
+soccerWord.match(goRegex); // ["goooooooo"]
+gPhrase.match(goRegex); // ["g"]
+oPhrase.match(goRegex); // null
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/using-the-test-method.md:
```diff
@@ -10,15 +10,14 @@ dashedName: using-the-test-method
 let testRegex = /Code/;
-testRegex.test(testStr);
-// Returns true
+testRegex.test(testStr); // true
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/use-capture-groups-to-search-and-replace.md:
```diff
@@ -15,20 +15,18 @@ You can search and replace text in a string using `.replace()` on a string. The
 let silverRegex = /silver/;
-wrongText.replace(silverRegex, "blue");
-// Returns "The sky is blue."
+wrongText.replace(silverRegex, "blue"); // "The sky is blue."
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/specify-upper-and-lower-number-of-matches.md:
```diff
@@ -12,19 +12,19 @@ Recall that you use the plus sign `+` to look for one or more characters and the
 let multipleA = /a{3,5}h/;
-multipleA.test(A4); // Returns true
-multipleA.test(A2); // Returns false
+multipleA.test(A4); // true
+multipleA.test(A2); // false
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/specify-only-the-lower-number-of-matches.md:
```diff
@@ -12,21 +12,21 @@ You can specify the lower and upper number of patterns with quantity specifiers
-multipleA.test(A4); // Returns true
-multipleA.test(A2); // Returns false
-multipleA.test(A100); // Returns true
+multipleA.test(A4); // true
+multipleA.test(A2); // false
+multipleA.test(A100); // true
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/specify-exact-number-of-matches.md:
```diff
@@ -12,21 +12,21 @@ You can specify the lower and upper number of patterns with quantity specifiers
-multipleHA.test(A4); // Returns false
-multipleHA.test(A3); // Returns true
-multipleHA.test(A100); // Returns false
+multipleHA.test(A4); // false
+multipleHA.test(A3); // true
+multipleHA.test(A100); // false
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/reuse-patterns-using-capture-groups.md:
```diff
@@ -19,8 +19,8 @@ The example below matches any word that occurs twice separated by a space:
 let repeatRegex = /(\w+)\s\1/;
-repeatRegex.test(repeatStr); // Returns true
-repeatStr.match(repeatRegex); // Returns ["regex regex", "regex"]
+repeatRegex.test(repeatStr); // true
+repeatStr.match(repeatRegex); // ["regex regex", "regex"]
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-ending-string-patterns.md:
```diff
@@ -15,21 +15,19 @@ You can search the end of strings using the dollar sign character `$` at the end
 let storyRegex = /story$/;
-storyRegex.test(theEnding);
-// Returns true
+storyRegex.test(theEnding); // true
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-ending-string-patterns.md:
```diff
@@ -15,21 +15,19 @@ You can search the end of strings using the dollar sign character `$` at the end
 let noEnding = "Sometimes a story will have to end";
-storyRegex.test(noEnding);
-// Returns false
+storyRegex.test(noEnding); // false
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-everything-but-letters-and-numbers.md:
```diff
@@ -16,8 +16,8 @@ You can search for the opposite of the `\w` with `\W`. Note, the opposite patter
 let sentence = "Coding!";
-numbers.match(shortHand); // Returns ["%"]
-sentence.match(shortHand); // Returns ["!"]
+numbers.match(shortHand); // ["%"]
+sentence.match(shortHand); // ["!"]
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-letters-of-the-alphabet.md:
```diff
@@ -19,9 +19,9 @@ let catStr = "cat";
-catStr.match(bgRegex); // Returns ["cat"]
-batStr.match(bgRegex); // Returns ["bat"]
-matStr.match(bgRegex); // Returns null
+catStr.match(bgRegex); // ["cat"]
+batStr.match(bgRegex); // ["bat"]
+matStr.match(bgRegex); // null
```
Remove comments from code segment.

RandellDawson replied:
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-literal-strings.md:
```diff
@@ -8,21 +8,20 @@ dashedName: match-literal-strings
 let testRegex = /Kevin/;
-testRegex.test(testStr);
-// Returns true
+testRegex.test(testStr); // true
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-literal-strings.md:
```diff
@@ -8,21 +8,20 @@ dashedName: match-literal-strings
 let wrongRegex = /kevin/;
 wrongRegex.test(testStr);
-// Returns false
+// false
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-non-whitespace-characters.md:
```diff
@@ -15,7 +15,7 @@ Search for non-whitespace using `\S`, which is an uppercase `s`. This pattern wi
 let whiteSpace = "Whitespace. Whitespace everywhere!"
 let nonSpaceRegex = /\S/g;
-whiteSpace.match(nonSpaceRegex).length; // Returns 32
+whiteSpace.match(nonSpaceRegex).length; // 32
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-single-character-with-multiple-possibilities.md:
```diff
@@ -12,18 +12,18 @@ You learned how to match literal patterns (`/literal/`) and wildcard character (
-bagStr.match(bgRegex); // Returns ["bag"]
-bugStr.match(bgRegex); // Returns ["bug"]
-bogStr.match(bgRegex); // Returns null
+bigStr.match(bgRegex); // ["big"]
+bagStr.match(bgRegex); // ["bag"]
+bugStr.match(bgRegex); // ["bug"]
+bogStr.match(bgRegex); // null
```
Remove comments from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:
```diff
@@ -23,16 +23,16 @@ let quit = "qu";
 let qRegex = /q(?!u)/;
-quit.match(quRegex); // Returns ["q"]
-noquit.match(qRegex); // Returns ["q"]
+quit.match(quRegex); // ["q"]
+noquit.match(qRegex); // ["q"]
```
Remove comment from code segment.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:
```diff
@@ -23,16 +23,16 @@ let quit = "qu";
 let password = "abc123";
 let checkPass = /(?=\w{3,6})(?=\D*\d)/;
-checkPass.test(password); // Returns true
+checkPass.test(password); // true
```
Remove comment from code segment.

RandellDawson reviewed (changes requested):
@nhcarrigan I went through all 211 files with changes and made comments on several changes I would like to see.

nhcarrigan commented:
> @nhcarrigan I went through all 211 files with changes and made comments on several changes I would like to see.

Perfect - thank you very much for your help with this. I'll start working through these today. 🙂

Commit:
fix: apply suggestions - thanks Randy! :)

Co-authored-by: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

Commit:
fix: non-suggestion comments

Commit:
chore(learn): remove comments from codes

Removes the comments from the description and instruction code
blocks to ensure that all relevant information is translatable.

nhcarrigan commented:
Okay @RandellDawson I believe I've extracted all of the comments from the description/instruction sections, with the exception of the JavaScript challenge that *introduces* comments.

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-nested-objects.md:
```diff
@@ -26,17 +26,19 @@ var ourStorage = {
 # --hints--
 
-`gloveBoxContents` should equal "maps".
+`gloveBoxContents` should equal `maps`.
```
```suggestion
`gloveBoxContents` should equal the string `maps`.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-bracket-notation.md:
````diff
@@ -40,7 +42,7 @@ Read the values of the properties `"an entree"` and `"the drink"` of `testObj` u
 ```
 
-The value of `entreeValue` should be `"hamburger"`
+The value of `entreeValue` should be `hamburger`
````
```suggestion
The value of `entreeValue` should be the string `hamburger`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-bracket-notation.md:
````diff
@@ -52,7 +54,7 @@ assert(entreeValue === 'hamburger');
 ```
 
-The value of `drinkValue` should be `"water"`
+The value of `drinkValue` should be `water`
````
```suggestion
The value of `drinkValue` should be the string `water`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-dot-notation.md:
````diff
@@ -36,7 +37,7 @@ Read in the property values of `testObj` using dot notation. Set the variable `h
 ```
 
-The value of `hatValue` should be `"ballcap"`
+The value of `hatValue` should be `ballcap`
````
```suggestion
The value of `hatValue` should be the string `ballcap`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-dot-notation.md:
````diff
@@ -48,7 +49,7 @@ assert(hatValue === 'ballcap');
 ```
 
-The value of `shirtValue` should be `"jersey"`
+The value of `shirtValue` should be `jersey`
````
```suggestion
The value of `shirtValue` should be the string `jersey`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-variables.md:
```diff
@@ -22,6 +22,8 @@ var myBreed = dogs[myDog];
 console.log(myBreed); // "Doberman"
```
```suggestion
console.log(myBreed);
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-variables.md:
````diff
@@ -32,10 +34,12 @@ function propPrefix(str) {
+console.log(someObj[someProp]);
 ```
 
+`someProp` has a value of the string `propName`, and the string `John` would be displayed in the console.
````
Again, the voice is changing now to `has`...

nhcarrigan replied:
`will have`? Or `holds`?

ShaunSHamilton replied:
I do not mind, just mentioning that it is changing. It can stay as is - I am nitpicky

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-variables.md:
````diff
@@ -56,7 +60,7 @@ The variable `player` should be a string
 ```
 
-The value of `player` should be "Montana"
+The value of `player` should be `Montana`
````
```suggestion
The value of `player` should be the string `Montana`
```

ShaunSHamilton reviewed (changes requested):
Most of these are nitpicky, but I wanted to bring it up... Also, a few more to come.

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.md:
```diff
@@ -31,38 +31,38 @@ switch (num) {
 # --hints--
 
-`switchOfStuff("a")` should have a value of "apple"
+`switchOfStuff("a")` should have a value of the string `apple`
```
```suggestion
`switchOfStuff("a")` should return the string `apple`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.md:
````diff
@@ -31,38 +31,38 @@ switch (num) {
 ```
 
-`switchOfStuff("b")` should have a value of "bird"
+`switchOfStuff("b")` should have a value of the string `bird`
````
```suggestion
`switchOfStuff("b")` should return the string `bird`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.md:
````diff
@@ -31,38 +31,38 @@ switch (num) {
 ```
 
-`switchOfStuff("c")` should have a value of "cat"
+`switchOfStuff("c")` should have a value of the string `cat`
````
```suggestion
`switchOfStuff("c")` should return the string `cat`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.md:
````diff
@@ -31,38 +31,38 @@ switch (num) {
 ```
 
-`switchOfStuff("d")` should have a value of "stuff"
+`switchOfStuff("d")` should have a value of the string `stuff`
````
```suggestion
`switchOfStuff("d")` should return the string `stuff`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.md:
````diff
@@ -31,38 +31,38 @@ switch (num) {
 ```
 
-`switchOfStuff(4)` should have a value of "stuff"
+`switchOfStuff(4)` should have a value of the string `stuff`
````
```suggestion
`switchOfStuff(4)` should return the string `stuff`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comment-your-javascript-code.md:
````diff
@@ -13,21 +13,20 @@ Comments are lines of code that JavaScript will intentionally ignore. Comments a
+Using `//` will tell JavaScript to ignore the remainder of the text on the current line. This is an in-line comment:
 
 ```js
 // This is an in-line comment.
````
Will these not be translated? If they will be, then I do not see the reason for the addition.

nhcarrigan replied:
They are in code blocks, so cannot be translated through Crowdin. When/if we update the comments dictionary to parse the description and instruction fields, these could be handled that way. But for now, there is no translation flow for them.

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-equality-operator.md:
```diff
@@ -22,34 +22,36 @@ function equalityTest(myVal) {
 # --instructions--
 
-Add the equality operator to the indicated line so that the function will return "Equal" when `val` is equivalent to `12`.
+Add the equality operator to the indicated line so that the function will return `Equal` when `val` is equivalent to `12`.
```
```suggestion
Add the equality operator to the indicated line so that the function will return the string `Equal` when `val` is equivalent to `12`.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-strict-equality-operator.md:
```diff
@@ -16,31 +16,33 @@ If the values being compared have different types, they are considered unequal,
 # --instructions--
 
-Use the strict equality operator in the `if` statement so the function will return "Equal" when `val` is strictly equal to `7`
+Use the strict equality operator in the `if` statement so the function will return `Equal` when `val` is strictly equal to `7`
```
```suggestion
Use the strict equality operator in the `if` statement so the function will return the string `Equal` when `val` is strictly equal to `7`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-strict-inequality-operator.md:
```diff
@@ -14,36 +14,38 @@ The strict inequality operator (`!==`) is the logical opposite of the strict equ
 # --instructions--
 
-Add the strict inequality operator to the `if` statement so the function will return "Not Equal" when `val` is not strictly equal to `17`
+Add the strict inequality operator to the `if` statement so the function will return `Not Equal` when `val` is not strictly equal to `17`
```
```suggestion
Add the strict inequality operator to the `if` statement so the function will return the string `Not Equal` when `val` is not strictly equal to `17`
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparisons-with-the-logical-or-operator.md:
```diff
@@ -36,7 +36,7 @@ return "Yes";
 # --instructions--
 
-Combine the two `if` statements into one statement which returns `"Outside"` if `val` is not between `10` and `20`, inclusive. Otherwise, return `"Inside"`.
+Combine the two `if` statements into one statement which returns `Outside` if `val` is not between `10` and `20`, inclusive. Otherwise, return `Inside`.
```
```suggestion
Combine the two `if` statements into one statement which returns the string `Outside` if `val` is not between `10` and `20`, inclusive. Otherwise, return the string `Inside`.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulate-arrays-with-shift.md:
````diff
@@ -18,9 +18,10 @@ Example:
-// removedFromOurArray now equals "Stimpson" and ourArray now equals ["J", ["cat"]].
 ```
 
+`removedFromOurArray` now equals `Stimpson` and ourArray now equals `["J", ["cat"]]`.
````
```suggestion
`removedFromOurArray` now equals the string `Stimpson`, and `ourArray` now equals `["J", ["cat"]]`.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/practice-comparing-different-values.md:
```diff
@@ -15,32 +15,30 @@ If the values being compared are not of the same type, the equality operator wil
 # --instructions--
 
-The `compareEquality` function in the editor compares two values using the equality operator. Modify the function so that it returns "Equal" only when the values are strictly equal.
+The `compareEquality` function in the editor compares two values using the equality operator. Modify the function so that it returns `Equal` only when the values are strictly equal.
```
```suggestion
The `compareEquality` function in the editor compares two values using the equality operator. Modify the function so that it returns the string `Equal` only when the values are strictly equal.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/quoting-strings-with-single-quotes.md:
```diff
@@ -26,9 +26,11 @@ However, this becomes a problem if you need to use the outermost quotes within i
+
 In the <dfn>goodStr</dfn> above, you can use both quotes safely by using the backslash <code>\\</code> as an escape character.
 
 **Note:** The backslash <code>\\</code> should not be confused with the forward slash `/`. They do not do the same thing.
```
Can this be changed, as it relates to the _work-around_ from Crowdin which I believe is now fixed?

nhcarrigan replied:
Ah yes, thank you.

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions.md:
````diff
@@ -22,7 +22,7 @@ function myFun() {
 ```
 
-The above outputs "Hello" to the console, returns "World", but `"byebye"` is never output, because the function exits at the `return` statement.
+The above outputs `Hello` to the console, returns `World`, but `byebye` is never output, because the function exits at the `return` statement.
````
```suggestion
The above outputs the string `Hello` to the console, returns the string `World`, but the string `byebye` is never output, because the function exits at the `return` statement.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/store-multiple-values-in-one-variable-using-javascript-arrays.md:
```diff
@@ -13,7 +13,7 @@ With JavaScript `array` variables, we can store several pieces of data in one pl
 You start an array declaration with an opening square bracket, end it with a closing square bracket, and put a comma between each entry, like this:
 
-`var sandwich = ["peanut butter", "jelly", "bread"]`.
+`var sandwich = ["peanut butter", "jelly", "bread"]`
```
Off-topic: Should this not be within triple backticks with a code language?

RandellDawson replied:
It could be, but as long as it only contains a single back tick code block, it will work.

ShaunSHamilton reviewed (changes requested):
Some more...

nhcarrigan commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/access-an-arrays-contents-using-bracket-notation.md:
````diff
@@ -16,47 +16,47 @@ When we define a simple array as seen below, there are 3 items in it:
 ```
 
-Using bracket notation, we have now reset the item at index 1 from `"b"`, to `"not b anymore"`.
+Using bracket notation, we have now reset the item at index 1 from `"b"`, to `"not b anymore"`. Now `ourArray` is `["a", "not b anymore", "c"]`.
````
IMHO, removing the quotes means we don't get confusion when we reference a string that does have quotes. Additionally, the quotation symbol differs between languages (I think).

ShaunSHamilton replied:
> Additionally, the quotation symbol differs between languages (I think).

That is a pain. I do think there is a point where there needs to be a standard used - programming languages' use of quotations, especially, and, thereby, the way they are mentioned. _Alas, my frustration._ 😞

nhcarrigan commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/access-multi-dimensional-arrays-with-indexes.md:
```diff
@@ -20,13 +20,14 @@ var arr = [
-There shouldn't be any spaces between the array name and the square brackets, like `array [0][0]` and even this `array [0] [0]` is not allowed. Although JavaScript is able to process this correctly, this may confuse other programmers reading your code.
+`arr[3]` would be `[[10, 11, 12], 13, 14]`, `arr[3][0]` would be `[10, 11, 12]`, and `arr[3][0][1]` would be `11`.
+
+**Note:** There shouldn't be any spaces between the array name and the square brackets, like `array [0][0]` and even this `array [0] [0]` is not allowed. Although JavaScript is able to process this correctly, this may confuse other programmers reading your code.
```
I could remove them, though that should be a separate PR I think to ensure this PR remains within the scope of formatting changes.

Commit:
fix: Apply suggestions from code review

Co-authored-by: Shaun Hamilton <51722130+ShaunSHamilton@users.noreply.github.com>

Commit:
fix: revert crowdin fix

nhcarrigan commented:
Thank you @ShaunSHamilton for the thorough review - I think I've caught the changes.

RandellDawson reviewed (changes requested):
@nhcarrigan I have one for sure suggested change but have a more general suggestion (similar to @ShaunSHamilton's comment, that the voice should be the same.  For example, I see the following intermixed throughout the changes:

Related to displaying things in the console:
- `would be displayed`
- `will display`

Related to functions returning values
- `would return`
- `return`
- `returns`
- `will return`

It may be the reason you chose a specific voice due to the surrounding text content's voice.  If not, then maybe they should all have the same voice?

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.md:
```diff
@@ -42,9 +44,11 @@ const usersObj = users.reduce((obj, user) => {
+
 # --instructions--
 
 The variable `watchList` holds an array of objects with information on several movies. Use `reduce` to find the average IMDB rating of the movies **directed by Christopher Nolan**. Recall from prior challenges how to `filter` data and `map` over it to pull what you need. You may need to create other variables, and return the average rating from `getRating` function. Note that the rating values ar ...
```
Do we want `directed by Christopher Nolan` translated?

RandellDawson replied:
Maybe `Christopher Nolan` should be in code blocks and get rid of the strong tags.

nhcarrigan replied:
Good idea.

Commit:
Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/mutations.md

Co-authored-by: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

nhcarrigan commented:
I'll give this another sweep and unify the voice 👍🏻

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-characters-that-occur-zero-or-more-times.md:
```diff
@@ -17,14 +17,16 @@ let soccerWord = "gooooooooal!";
 # --instructions--
 
-For this challenge, `chewieQuote` has been initialized as "Aaaaaaaaaaaaaaaarrrgh!" behind the scenes. Create a regex `chewieRegex` that uses the `*` character to match an uppercase `"A"` character immediately followed by zero or more lowercase `"a"` characters in `chewieQuote`. Your regex does not need flags or character classes, and it should not match any of the other quotes.
+For this challenge, `chewieQuote` has been initialized as "Aaaaaaaaaaaaaaaarrrgh!" behind the scenes. Create a regex `chewieRegex` that uses the `*` character to match an uppercase `A` character immediately followed by zero or more lowercase `a` characters in `chewieQuote`. Your regex does not need flags or character classes, and it should not match any of the other quotes.
```
```suggestion
For this challenge, `chewieQuote` has been initialized as the string `Aaaaaaaaaaaaaaaarrrgh!` behind the scenes. Create a regex `chewieRegex` that uses the `*` character to match an uppercase `A` character immediately followed by zero or more lowercase `a` characters in `chewieQuote`. Your regex does not need flags or character classes, and it should not match any of the other quotes.
```

ShaunSHamilton reviewed (changes requested):
Last of me

Commit:
fix: Apply suggestions from code review

Co-authored-by: Shaun Hamilton <51722130+ShaunSHamilton@users.noreply.github.com>

Commit:
Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-to-assign-variables-from-arrays.md

Co-authored-by: Shaun Hamilton <51722130+ShaunSHamilton@users.noreply.github.com>

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/understanding-undefined-value-returned-from-a-function.md:
````diff
@@ -18,9 +18,11 @@ var sum = 0;
+addSum(3);
 ```
 
+Here, `sum` will be modified but the returned value is `undefined`.
+
 `addSum` is a function without a `return` statement. The function will change the global `sum` variable but the returned value of the function is `undefined`.
````
```suggestion
`addSum` is a function without a `return` statement. The function will change the global `sum` variable but the returned value of the function is `undefined`.
```

ShaunSHamilton reviewed (changes requested):
Thank you, for all of this, Nicholas. Last one from me, just because _less is more better_.

RandellDawson commented:
> @nhcarrigan I have one for sure suggested change but have a more general suggestion (similar to @ShaunSHamilton's comment, that the voice should be the same. For example, I see the following intermixed throughout the changes:
> 
> Related to displaying things in the console:
> 
> * `would be displayed`
> * `will display`
> 
> Related to functions returning values
> 
> * `would return`
> * `return`
> * `returns`
> * `will return`
> 
> It may be the reason you chose a specific voice due to the surrounding text content's voice. If not, then maybe they should all have the same voice?

@nhcarrigan Regarding the changes in the 17th commit (where you created descriptions for the console.logs and removed the comments from the code segments), I do  not see any more commits where you unified the voice for these.  Did I miss something?

nhcarrigan commented:
@RandellDawson working on this at the moment :)

ojeytonwilliams reviewed (dismissed):
This is only a partial review, but it's looking good.  Only a few nitpicks.

Commit:
fix: Apply suggestions from code review

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>
Co-authored-by: Shaun Hamilton <51722130+ShaunSHamilton@users.noreply.github.com>

Commit:
chore: change voice

Commit:
fix: Christopher Nolan

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/remember-to-set-the-constructor-property-when-changing-the-prototype.md:
````diff
@@ -11,11 +11,13 @@ dashedName: remember-to-set-the-constructor-property-when-changing-the-prototype
+duck instanceof Bird;
 ```
 
+In order, these would return `false`, `true`, and `true`.
````
I am wondering when we have a sequence of expressions, if it would be better to say:
> In order, the above expressions would evaluate to `false`, `true`, and `true`.

There are a few above this one for this commit that I did not highlight.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-own-properties.md:
````diff
@@ -31,9 +31,11 @@ for (let property in duck) {
+console.log(ownProps);
 ```
 
+The console will display the value `["name", "numLegs"]`.
````
"would" or "will"?

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-constructor-property.md:
````diff
@@ -14,10 +14,12 @@ There is a special `constructor` property located on the object instances `duck`
+console.log(beagle.constructor === Dog);
 ```
 
+Both of these `console.log` calls will display `true` in the console.
````
"would" or "will"?

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-a-mixin-to-add-common-behavior-between-unrelated-objects.md:
````diff
@@ -40,10 +40,12 @@ flyMixin(plane);
+plane.fly();
 ```
 
+The console will display the string `Flying, wooosh!` twice, once for each `.fly()` call.
````
"would" or "will"?

RandellDawson reviewed (commented):
@nhcarrigan A few more questions.

Commit:
fix: expressions would evaluate

Commit:
fix: will -> would

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/add-methods-after-inheritance.md:
````diff
@@ -38,7 +38,7 @@ duck.eat();
 ```
 
-`duck.eat()` will display the string `nom nom nom` in the console, and `duck.fly()` will display the string `I'm flying!`.
+`duck.eat()` wou;d display the string `nom nom nom` in the console, and `duck.fly()` would display the string `I'm flying!`.
````
```suggestion
`duck.eat()` would display the string `nom nom nom` in the console, and `duck.fly()` would display the string `I'm flying!`.
```

ieahleen commented:
Can I also suggest removing "to work" from
> Use a for loop to work to push the values 1 through 5 onto myArray.

it's in the file `curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.md`

Commit:
Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/add-methods-after-inheritance.md

Co-authored-by: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

Commit:
fix: to work to push

RandellDawson reviewed (dismissed):
LGTM

RandellDawson commented:
@shauncromie @ojeytonwilliams If you are OK with the latest version of this, we can merge it.

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.md:
```diff
@@ -11,19 +11,19 @@ dashedName: iterate-with-javascript-for-loops
+The condition statement is evaluated at the beginning of every loop iteration and will continue as long as it evaluates to `true`. When the condition is `false` at the start of the iteration, the loop will stop executing. This means if the condition starts as false, your loop will never execute.
 
-The `final-expression` is executed at the end of each loop iteration, prior to the next `condition` check and is usually used to increment or decrement your loop counter.
+The final expression is executed at the end of each loop iteration, prior to the next `condition` check and is usually used to increment or decrement your loop counter.
```
```suggestion
The final expression is executed at the end of each loop iteration, prior to the next condition check and is usually used to increment or decrement your loop counter.
```

ShaunSHamilton commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/add-methods-after-inheritance.md:
```diff
@@ -34,13 +34,15 @@ Now instances of `Bird` will have both `eat()` and `fly()` methods:
 # --instructions--
 
-Add all necessary code so the `Dog` object inherits from `Animal` and the `Dog's` `prototype` constructor is set to Dog. Then add a `bark()` method to the `Dog` object so that `beagle` can both `eat()` and `bark()`. The `bark()` method should print "Woof!" to the console.
+Add all necessary code so the `Dog` object inherits from `Animal` and the `Dog's` `prototype` constructor is set to Dog. Then add a `bark()` method to the `Dog` object so that `beagle` can both `eat()` and `bark()`. The `bark()` method should print `Woof!` to the console.
```
This is more out of my personal interest:
- How does something like this translate in other languages:
  > ...the `Dog's` `prototype` constructor is set to Dog...

Specifically, I am wondering about the apostrophe in the backticks, and the word "Dog" at the end not.

RandellDawson replied:
Maybe it should be:
```suggestion
Add all necessary code so the `Dog` object inherits from `Animal` and the `Dog`'s `prototype` constructor is set to `Dog`. Then add a `bark()` method to the `Dog` object so that `beagle` can both `eat()` and `bark()`. The `bark()` method should print `Woof!` to the console.
```

ShaunSHamilton reviewed (changes requested):
I tried my best not to be nitpicky, but I do have 2 concerns

Commit:
Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-for-loops.md

Co-authored-by: Shaun Hamilton <51722130+ShaunSHamilton@users.noreply.github.com>

Commit:
Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/add-methods-after-inheritance.md

Co-authored-by: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

ShaunSHamilton reviewed (approved):
This LGTM 👍 
I am not envious of you having to go through that, Nicholas. Well done.

RandellDawson reviewed (approved):
@nhcarrigan LGTM.  Do you want to wait for @ojeytonwilliams to approve?

nhcarrigan commented:
I'm fine either way, though landing this would be nice (we've racked up so many review suggestions that it takes a few minutes for me to load lol)

Thank you all for the help with getting this correct, too. :)

RandellDawson commented:
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉
